src/components/TodoItem.tsx

```diff
@@ -0,0 +1,99 @@
+import cn from 'classnames';
+
+import {
+  useContext,
+  useState,
+  useRef,
+  useEffect,
+} from 'react';
+import { Todo } from '../types';
+import { TodosContext } from '../TodosContext';
+
+interface Props {
+  todo: Todo
+}
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { toggleTodo, removeTodo, editTodo } = useContext(TodosContext);
+  const { id, title, completed } = todo;
+
+  const [editingMode, setEditingMode] = useState(false);
```

Comment:
Rename all boolean variables using [tips](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)

```suggestion
  const [isEditingMode, setIsEditingMode] = useState(false);
```